parser/parse_uri.go

```diff
@@ -27,11 +38,11 @@ func ParseUri(uriStr string, uri *sip.Uri) (err error) {
 func uriStateSIP(uri *sip.Uri, s string) (uriFSM, string, error) {
 	switch s[0] {
 	case 'S', 's':
-		if s[3] == 'S' || s[3] == 's' {
```

Comment:
I think that maybe we should require sip/sips to be present, per comment above. This maybe would then simplify check

Reply:
Updated the code to check for `sip:` and `sips:`.

Reply:
ok this maybe now decreases performance. 
We need also to check upper Case. SIP, SIPS, that is why old implementation was looking both. 
I mostly wanted to remove default case, but keep this small check. 
This is how is done in some other implements, compromise performance and full check.

Reply:
> We need also to check upper Case. SIP, SIPS, that is why old implementation was looking both.

`strings.EqualFold` does case-insensitive matching. I added test cases to verify that we match both `sip:` and `SIP:`.

I don't think a huge amount of performance is lost here.